File: 801-900/898.py
# Author: Huahua
# Running time: 812 ms
class Solution:
  def subarrayBitwiseORs(self, A):
    cur = set()
    ans = set()
    for a in A:
      cur = {a | b for b in cur} | {a}
      ans |= cur      
    return len(ans)

File: 801-900/898.java
// Author: Huahua
// Running time: 434 ms
class Solution {
  public int subarrayBitwiseORs(int[] A) {
    Set<Integer> ans = new HashSet<>();
    Set<Integer> cur = new HashSet<>();
    for (int a : A) {
      Set<Integer> nxt = new HashSet<>();
      nxt.add(a);
      for (int b : cur)
        nxt.add(a | b);
      ans.addAll(nxt);
      cur = nxt;
    }
    return ans.size();
  }
}
